omdb/src/bin/omdb/sled.rs

```diff
@@ -0,0 +1,109 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! omdb commands that query or update specific Sleds
+
+use anyhow::bail;
+use anyhow::Context;
+use clap::Args;
+use clap::Subcommand;
+
+/// Arguments to the "omdb sled" subcommand
+#[derive(Debug, Args)]
+pub struct SledArgs {
+    /// URL of the Sled internal API
+    #[clap(long, env("OMDB_SLED_URL"))]
+    sled_url: Option<String>,
+
+    #[command(subcommand)]
+    command: SledCommands,
+}
+
+/// Subcommands for the "omdb sled" subcommand
+#[derive(Debug, Subcommand)]
+enum SledCommands {
+    /// print information about zones
+    #[clap(subcommand)]
+    Zones(ZoneCommands),
+
+    /// print information about zpools
+    #[clap(subcommand)]
+    Zpools(ZpoolCommands),
+}
+
+#[derive(Debug, Subcommand)]
+enum ZoneCommands {
+    /// Print list of all zones
```

Comment:
```suggestion
    /// Print list of control plane zones that the sled agent is configured to run
```

(just trying to be precise about what it's printing so that someone doesn't interpret this as the same as `zoneadm list`)

edit: I just checked and it looks like this is only the *running* zones that are the subset of zones the sled agent is configured to run, I think.

Reply:
Updated, specifying that these are "running control plane zones". I think we could **also** expose the set of zones the sled was **told** to run, but this will require modifying the sled agent API.